dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -6,21 +6,58 @@
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
+    winrt::Windows::Foundation::TypedEventHandler<
+        Microsoft::Windows::PushNotifications::PushNotificationChannel,
+        Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> s_typedEventHandler;
+
     PushNotificationChannel::PushNotificationChannel(winrt::Windows::Networking::PushNotifications::PushNotificationChannel const& channel)
     {
         m_channel = channel;
     }
     winrt::Windows::Foundation::Uri PushNotificationChannel::Uri()
     {
+        auto lock = m_lock.lock();
         winrt::Windows::Foundation::Uri pushChannelUri{ m_channel.Uri() };
         return pushChannelUri;
     }
     winrt::Windows::Foundation::DateTime PushNotificationChannel::ExpirationTime()
     {
+        auto lock = m_lock.lock();
         return m_channel.ExpirationTime();
     }
     void PushNotificationChannel::Close()
     {
+        auto lock = m_lock.lock();
         m_channel.Close();
     }
+
+    winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::Foundation::TypedEventHandler<Microsoft::Windows::PushNotifications::PushNotificationChannel, Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> const& handler)
+    {
+        winrt::Windows::Foundation::TypedEventHandler<
+            winrt::Windows::Networking::PushNotifications::PushNotificationChannel,
+            winrt::Windows::Networking::PushNotifications::PushNotificationReceivedEventArgs> typedEventHandler
+            (this, &winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationChannel::LambdaWrapper);
```

Comment:
I would strongly recommend that you inline the LambdaWrapper here instead of defining it as a private function.

Reply:
I can do that. Just wondering, is this a standard coding pattern we should be doing with our lambda functions, or is there some inefficiency by defining it as a private function?

Reply:
Its just a common pattern we use for lambdas today. If the code is manageable or small enough, inlining is sufficient. More complex functions can have their own declaration.